src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Microsoft.OpenApi.Models;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    // Given Power Fx arguments, translate into a HttpRequestMessage and invoke.
+    internal class HttpFunctionInvoker
+    {
+        private readonly HttpMessageInvoker _httpClient;
+        private readonly HttpMethod _method;
+        private readonly string _path;
+        private readonly FormulaType _returnType;
+        private readonly ArgumentMapper _argMapper;
+        private readonly ICachingHttpClient _cache;
+
+        public HttpFunctionInvoker(
+            HttpMessageInvoker httpClient, 
+            HttpMethod method, 
+            string path, 
+            FormulaType returnType, 
+            ArgumentMapper argMapper,
+            ICachingHttpClient cache = null)
+        {
+            _httpClient = httpClient;
+            _method = method;
+            _path = path;
+            _argMapper = argMapper;
+            _cache = cache ?? NonCachingClient.Instance;
+            _returnType = returnType;
+        }
+
+        internal static void VerifyCanHandle(ParameterLocation? location)
+        {
+            switch (location.Value)
+            {
+                case ParameterLocation.Path:
+                case ParameterLocation.Query:
+                    break;
+
+                default:
+                    throw new NotImplementedException($"Unsupported ParameterIn {location}");
+            }
+        }
+
+        public HttpRequestMessage BuildRequest(FormulaValue[] args)
+        {
+            var path = _path;
+            var query = new StringBuilder();
+
+            var map = _argMapper.ConvertToSwagger(args);
+            foreach (var param in _argMapper._openApiParameters)
+            {
+                if (map.TryGetValue(param.Name, out var paramValue))
+                {
+                    var valueStr = paramValue.ToObject().ToString();
+
+                    switch (param.In.Value)
+                    {
+                        case ParameterLocation.Path:
+                            path = path.Replace("{" + param.Name + "}", HttpUtility.UrlEncode(valueStr));
```

Comment:
Should we provide the Encoding for UrlEncode?

Reply:
can you elaborate?

Reply:
[https://docs.microsoft.com/en-us/dotnet/api/system.web.httputility.urlencode?view=net-6.0#system-web-httputility-urlencode(system-string-system-text-encoding](https://docs.microsoft.com/en-us/dotnet/api/system.web.httputility.urlencode?view=net-6.0#system-web-httputility-urlencode(system-string-system-text-encoding)) points to [https://docs.microsoft.com/en-us/previous-versions/aspnet/a2a4yykt(v=vs.100](https://docs.microsoft.com/en-us/previous-versions/aspnet/a2a4yykt(v=vs.100))

Reply:
The key thing is that we're calling UrlEncode at all.

Are you talking about passing the 2nd encoding parameter here as opposed to using the default?